### PR TITLE
Call ExoBLAS reusable workflow

### PR DESCRIPTION
**Pull request by SamirDroubi**

### CODE REVIEW

File: .github/workflows/ExoBLAS.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  call_test_ExoBLAS:
+    # point to reusable-workflow branch for testing before merging
+    uses: exo-lang/ExoBLAS/.github/workflows/test_ExoBLAS.yml@reusable-workflow
+    with:
+      exo-ref: ${{ github.event.pull_request.head.ref }}"
```
Comment (yamaguchi1024) [inline .github/workflows/ExoBLAS.yml:13]: How does `github.event.pull_request.head.ref` gonna work on main?

Comment (SamirDroubi) [reply .github/workflows/ExoBLAS.yml:13]: Yeah I need to check what triggered the action and decide the ref based on it. But I was trying to get it to work on the pull request ref for starters.

Comment (yamaguchi1024) [reply .github/workflows/ExoBLAS.yml:13]: I see that makes sense, lmk when you want to merge this

**Comment by SamirDroubi** [comment]
Still need to change the branch protection rules, but I will do that once this is ready to be merged.

**Review by yamaguchi1024 — APPROVED** [review]

**Pull request merged**